### PR TITLE
[8.1] Skip unavailable storages in scanner

### PR DESCRIPTION
**Pull request by icewind1991**
Backport of https://github.com/owncloud/core/pull/20789

### CODE REVIEW

File: lib/private/files/utils/scanner.php [hunk 1]
```diff
@@ -156,7 +165,13 @@ public function scan($dir = '') {
 			$scanner->setUseTransactions(false);
 			$this->attachListener($mount);
 			$this->db->beginTransaction();
-			$scanner->scan($relativePath, \OC\Files\Cache\Scanner::SCAN_RECURSIVE, \OC\Files\Cache\Scanner::REUSE_ETAG | \OC\Files\Cache\Scanner::REUSE_SIZE);
+			try {
+				$scanner->scan($relativePath, \OC\Files\Cache\Scanner::SCAN_RECURSIVE, \OC\Files\Cache\Scanner::REUSE_ETAG | \OC\Files\Cache\Scanner::REUSE_SIZE);
+			} catch (StorageNotAvailableException $e) {
+				$this->logger->error('Storage ' . $storage->getId() . ' not available');
+				$this->logger->logException($e);
```
Comment (MorrisJobke) [inline lib/private/files/utils/scanner.php]: LogException is not available in stable8.1:

```
PHP Fatal error:  Call to undefined method OC\Log::logException() in /Users/morrisjobke/Projects/owncloud/master/lib/private/files/utils/scanner.php on line 172
```

Comment (icewind1991) [inline lib/private/files/utils/scanner.php]: fixed

**Comment by PVince81** [comment]
:+1:

**Comment by MorrisJobke** [comment]
I rebased to retrigger CI

**Comment by MorrisJobke** [comment]
Tested and still works :+1: 


@DeepDiver1975 Are those failing tests correct? This is stable and PHP7 & OCS tests aren't working there, but how about the docker container?

**Comment by DeepDiver1975** [comment]
needs rebase

failing smashbox tests are fishy

**Comment by MorrisJobke** [comment]
> needs rebase
> 
> failing smashbox tests are fishy

Done.

**Comment by PVince81** [comment]
Smashbox tests passed, merging.

**Comment by PVince81** [comment]
Wait, no.

These ones failed:
smashbox-on-docker-ci/DOCKER_IMAGE=ubuntu_oc_lamp-git,TEST_N

**Comment by PVince81** [comment]
The failure is likely unrelated:
```
2016-01-07 07:00:59,302 - ERROR - owner - recipient 4 uploads /test/test4.txt through reshare etag for / previous ["568e5376dac26"] new ["568e5376dac26"] 'etag for / previous [%s] new [%s]' % (root_etag6, root_etag7)) failed in owner() ["/ssd/jenkins/workspace/smashbox-on-docker-ci/DOCKER_IMAGE/ubuntu_oc_lamp-git/TEST_NAME/test_sharePropagationGroups/mirallBranch/v2.0.2/slave/SMASH/smashbox/lib/owncloud/test_sharePropagationGroups.py" at line 179]
```

Maybe we didn't backport the fix for when group names are the same as user names.

**Comment by PVince81** [comment]
Hmm, I thought this was https://github.com/owncloud/core/issues/20892 but it's slightly different.
Let me check if it fails on 8.1 too.

**Comment by PVince81** [comment]
These tests fail on stable8.1 too, raised here https://github.com/owncloud/core/issues/21541

**Pull request merged**